be/src/service/service_be/lake_service.cpp

```diff
@@ -147,30 +147,21 @@ void LakeServiceImpl::publish_version(::google::protobuf::RpcController* control
         return;
     }
 
-    auto enable_trace = config::lake_enable_publish_version_trace_log;
     auto start_ts = butil::gettimeofday_us();
     auto thread_pool = publish_version_thread_pool(_env);
     auto latch = BThreadCountDownLatch(request->tablet_ids_size());
     bthread::Mutex response_mtx;
-    Trace* trace = nullptr;
-    scoped_refptr<Trace> trace_gurad;
-
-    if (enable_trace) {
-        trace_gurad = scoped_refptr<Trace>(new Trace());
-        trace = trace_gurad.get();
-        TRACE_TO(trace, "got request. txn_id=$0 new_version=$1 #tablets=$2", request->txn_ids(0),
-                 request->new_version(), request->tablet_ids_size());
-    }
+    scoped_refptr<Trace> trace_gurad = scoped_refptr<Trace>(new Trace());
+    Trace* trace = trace_gurad.get();
+    TRACE_TO(trace, "got request. txn_id=$0 new_version=$1 #tablets=$2", request->txn_ids(0), request->new_version(),
+             request->tablet_ids_size());
 
     for (auto tablet_id : request->tablet_ids()) {
         auto task = [&, tablet_id]() {
             DeferOp defer([&] { latch.count_down(); });
-            Trace* sub_trace = nullptr;
-            if (enable_trace) {
-                scoped_refptr<Trace> child_trace(new Trace);
-                sub_trace = child_trace.get();
-                trace->AddChildTrace("PublishTablet", sub_trace);
-            }
+            scoped_refptr<Trace> child_trace(new Trace);
+            Trace* sub_trace = child_trace.get();
+            trace->AddChildTrace(std::to_string(tablet_id), sub_trace);
```

Comment:
In the current implementation, the first argument of `AddChildTrace` must be string literal, otherwise there will be illegal memory accesses.

Reply:
So I'm afraid that either you'll have to modify the label here so that it's a constant, or you'll have to modify Trace's implementation to allow for non-constant labels.

Reply:
Got it.

Reply:
I will trace tablet_id in sub trace.

Reply:
Done